Source/santasyncservice/SNTSyncRuleDownload.mm

```diff
@@ -42,7 +43,17 @@ SNTRuleCleanup SyncTypeToRuleCleanup(SNTSyncType syncType) {
   }
 }
 
-@implementation SNTSyncRuleDownload
+@implementation SNTSyncRuleDownload {
+  std::unique_ptr<santa::cel::Evaluator> _celEvaluator;
+}
+
+- (nullable instancetype)initWithState:(nonnull SNTSyncState *)syncState {
+  self = [super initWithState:syncState];
+  if (self) {
+    _celEvaluator = santa::cel::Evaluator::Create().value();
```

Comment:
This feels a bit scary since it will throw if no value. But it seems like maybe there isn't much else for us to do. It looks like failures in `Create` would either be transitive, or always fail. Biggest concern is a crash loop, but that feels like something that would be caught quickly since it would mean the sync service never completes rule download...

Do we need an evaluator per rule download instance? Or could we instead create one for the lifetime of the sync service? If we create one for the lifetime, we could make that a requirement for loading which would move any failures forward and be more easily discovered via our testing.

Reply:
> Do we need an evaluator per rule download instance? Or could we instead create one for the lifetime of the sync service?

No, creating at process start and passing it into stages via SNTSyncState is reasonable and what I've gone for. That did involve moving most of the remaining ObjC classes to ObjC++.

> If we create one for the lifetime, we could make that a requirement for loading which would move any failures forward and be more easily discovered via our testing.

I'm leaving the evaluator as a nullable pointer with a check around it, I don't think we should break syncing if we can't create an evaluator.